Server/ConcordiaCurriculumManager/Controllers/GroupController.cs

```diff
@@ -127,4 +127,35 @@ public async Task<IActionResult> RemoveGroupMaster([FromRoute, Required] Guid gr
 
         return Ok();
     }
+
+    [HttpPut("{groupId}")]
+    [Authorize(Policies.IsGroupMasterOrAdmin)]
+    [SwaggerResponse(StatusCodes.Status200OK, "Group updated successfully")]
+    [SwaggerResponse(StatusCodes.Status404NotFound, "Group does not exist")]
+    public async Task<IActionResult> UpdateGroup(Guid id, [FromBody] GroupCreateDTO groupDto)
```

Comment:
Both these endpoints seem to fail because you have the parameter in the route be `groupId`, but then in the method signature you rename it to `id`, so the route parameter doesn't map correctly.